### PR TITLE
[kube-prometheus-stack] allow overriding of selector for kube-state-metrics serviceMonitor

### PR DESCRIPTION
**Pull request by janlud**
`app.kubernetes.io/instance` is overriden by ArgoCD on the service and because of that this `serviceMonitor` fails to detect the service

Signed-off-by: Jan Ludvik <jan.ludvik@deltatre.com>

#### What this PR does / why we need it:
Allows optionally overriding of kube-state-metrics serviceMonitor.
The reason is that ArgoCD overrides the `app.kubernetes.io/instance` of all deployment so this `serviceMonitor` cannot then detect anything to scrape.

Using this override will allow any to replace the selector with any value while keeping backwards compatibility with previous versions.

#### Special notes for your reviewer:
Using additional custom serviceMonitor in the chart but as the serviceMonitor doesn't have disable I have 2 in Prometheus.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by vsliouniaev — APPROVED** [review]

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 11.0.0
+version: 11.0.1
```
Comment (gkarthiks) [inline charts/kube-prometheus-stack/Chart.yaml:20]: upgrade the chart version

**Review by monotek — CHANGES_REQUESTED** [review]
Please merge master and raise minor version, as your pr add new feature.

**Comment by janlud** [comment]
Sorry it took so long! I updated my branch from master and raised the minor version as requested

**Review by monotek — APPROVED** [review]

**Review by monotek — DISMISSED** [review]

**Review by vsliouniaev — APPROVED** [review]

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 12.10.6
+version: 12.11.0
```
Comment (gkarthiks) [inline charts/kube-prometheus-stack/Chart.yaml:21]: please sync your fork to resolve the merge conflicts.

**Review by gkarthiks — COMMENTED** [review]
please bump the chart version

**Comment by monotek** [comment]
chart version updated

**Comment by gkarthiks** [comment]
@monotek so sorry. But can you please bump the chart version one more time? 🙏

**Comment by monotek** [comment]
done :-)
p.s. you should be ablte to do it too...

**Review by gkarthiks — APPROVED** [review]

**Pull request merged**